frontend/src/metabase/dashboard/components/DashCard/DashCard.tsx

```diff
@@ -130,33 +132,42 @@ function DashCardInner({
     }
   });
 
-  const mainCard: Card = useMemo(
+  const mainCard: Card | VirtualCard = useMemo(
     () => ({
       ...dashcard.card,
       visualization_settings: mergeSettings(
-        dashcard.card.visualization_settings,
+        dashcard?.card?.visualization_settings,
         dashcard.visualization_settings,
       ),
     }),
     [dashcard],
   );
 
   const cards = useMemo(() => {
-    if (Array.isArray(dashcard.series)) {
+    if ("series" in dashcard && Array.isArray(dashcard.series)) {
```

Comment:
We can use a type guard here:

```suggestion
    if (isQuestionDashCard(dashcard)) {
```

Reply:
Fixed in [`5e7d3eb`](https://github.com/metabase/metabase/pull/38444/commits/5e7d3eb41e042a876f46c402071f268233ba3acd)